src/main/java/com/autilus/hw0706/ListImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.autilus.hw0706;
+
+public class ListImpl<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+    private int size;
+    private int position;
+
+    private void checkCapasity(int i) {
+        if (i >= size) {
+            size += size >> 1;
+            array = copyToNewArray(array);
+        }
+    }
+
+    private Object[] copyToNewArray(Object[] newArray) {
+        newArray = new Object[size];
+        System.arraycopy(array, 0, newArray, 0, position);
+        return newArray;
+    }
+
+    @Override
+    public void add(T value) {
+        checkCapasity(position);
+        add(value, position);
+        position++;
+    }
+
+    @Override
+    public void add(T value, int index) {
+        checkCapasity(index);
+        size++;
+        array = copyToNewArray(array);
+        System.arraycopy(array, index, array, index + 1, size - index - 1);
+        array[index] = value;
+    }
+
+    @Override
+    public void addAll(List list) {
+        Object[] objects = list.toArray();
+        size = objects.length;
+        array = copyToNewArray(array);
+        System.arraycopy(objects, 0, array, 0, objects.length);
+        position += this.size;
+    }
+
+    @Override
+    public T get(int index) {
+        if (index >= size) {
+            return null;
+        }
+        return (T) array[index];
+    }
+
+    @Override
+    public void set(T value, int index) {
+        if (index >= size) {
+            throw new ArrayIndexOutOfBoundsException("can't set nonexistent index");
+        }
+        array[index] = value;
+    }
+
+    @Override
+    public T remove(int index) {
+        size--;
+        System.arraycopy(array, index, array, index - 1, size - index - 1);
+        return (T) array;
```

Comment:
we need return one value here

Reply:
what we need to return?

Reply:
we need return the value which we remove. Don't all elements in array)